cockatrice/src/game/filters/filter_string.cpp

```diff
@@ -261,14 +268,22 @@ static void setupParserRules()
         return QString::fromStdString(std::string(sv.sv()));
     };
 
-    search["RegexString"] = [](const peg::SemanticValues &sv) -> StringMatcher {
+    search["NormalMatcher"] = [](const peg::SemanticValues &sv) -> StringMatcher {
         auto target = std::any_cast<QString>(sv[0]);
-        return [=](const QString &s) {
-            auto sanitizedTarget = QString(target);
-            sanitizedTarget.replace("\\\"", "\"");
-            sanitizedTarget.replace("\\'", "'");
-            return s.contains(sanitizedTarget, Qt::CaseInsensitive);
-        };
+        auto sanitizedTarget = QString(target);
+        sanitizedTarget.replace("\\\"", "\"");
+        sanitizedTarget.replace("\\'", "'");
+        return [=](const QString &s) { return s.contains(sanitizedTarget, Qt::CaseInsensitive); };
+    };
+
+    search["RegexMatcher"] = [](const peg::SemanticValues &sv) -> StringMatcher {
+        auto target = std::any_cast<QString>(sv[0]);
+        auto regex = QRegularExpression(target, QRegularExpression::CaseInsensitiveOption);
```

Comment:
how well does this perform when the user say, decides to add a lot of recursive regexes?

Reply:
Not sure this is a huge concern for me tbh, if it's only locally that things can go wrong. This is a power use feature, regardless, so these types of users should know to be careful